docker/docker-compose.yml

```diff
@@ -10,93 +10,136 @@ etcd:
+  - DOCKER_DAEMON_ARGS="--log-level=error"
+  links:
+  - etcd
+  - mesosmaster1
+  - "ambassador:apiserver"
+  volumes:
+  - /var/run/docker.sock:/var/run/docker.sock
+mesosslave2:
+  hostname: mesosslave2
+  privileged: true
+  image: mesosphere/kubernetes-mesos-slave:latest
+  ports: [ "5052:5052" ]
+  environment:
+  - MESOS_HOSTNAME=mesosslave2
+  - MESOS_MASTER=mesosmaster1:5050
+  - MESOS_PORT=5052
+  - MESOS_LOG_DIR=/var/log/mesos
+  - MESOS_LOGGING_LEVEL=INFO
+  - MESOS_RESOURCES=cpus:4;mem:1280;disk:25600;ports:[22000-22099]
+  - MESOS_SWITCH_USER=0
+  - MESOS_CONTAINERIZERS=docker,mesos
+  - DOCKER_DAEMON_ARGS="--log-level=error"
+  links:
+  - etcd
+  - mesosmaster1
+  - "ambassador:apiserver"
   volumes:
   - /var/run/docker.sock:/var/run/docker.sock
-#  - zookeeper
 apiserver:
   hostname: apiserver
-  image: mesosphere/kubernetes-mesos:latest
+  image: mesosphere/kubernetes-mesos
   entrypoint:
   - /bin/bash
   - "-c"
   - >
     echo "Hostname: $(hostname -f) ($(hostname -f | xargs resolveip))" &&
-    (grep "mesos-master\s*=" /opt/mesos-cloud.conf || echo "  mesos-master = mesos:5050" >> /opt/mesos-cloud.conf) &&
+    (grep "mesos-master\s*=" /opt/mesos-cloud.conf || echo "  mesos-master = mesosmaster1:5050" >> /opt/mesos-cloud.conf) &&
     await-health-check http://etcd:4001/health &&
-    await-health-check http://mesos:5050/health &&
+    await-health-check http://mesosmaster1:5050/health &&
     km apiserver
     --address=$(resolveip apiserver)
-    --external_hostname=apiserver
-    --etcd_servers=http://etcd:4001
-    --portal_net=10.10.10.0/24
+    --external-hostname=apiserver
+    --etcd-servers=http://etcd:4001
+    --portal-net=10.10.10.0/24
     --port=8888
-    --cloud_provider=mesos
-    --cloud_config=/opt/mesos-cloud.conf
+    --service-node-port-range=22075-22125
+    --cloud-provider=mesos
+    --cloud-config=/opt/mesos-cloud.conf
     --v=2
   ports: [ "8888:8888" ]
   links:
   - etcd
-  - mesos
+  - mesosmaster1
+  - mesosslave1
+  - mesosslave2
 controller:
   hostname: controller
-  image: mesosphere/kubernetes-mesos:latest
+  image: mesosphere/kubernetes-mesos
   entrypoint:
   - /bin/bash
   - "-c"
   - >
     echo "Hostname: $(hostname -f) ($(hostname -f | xargs resolveip))" &&
-    (grep "mesos-master\s*=" /opt/mesos-cloud.conf || echo "  mesos-master = mesos:5050" >> /opt/mesos-cloud.conf) &&
+    (grep "mesos-master\s*=" /opt/mesos-cloud.conf || echo "  mesos-master = mesosmaster1:5050" >> /opt/mesos-cloud.conf) &&
     await-health-check http://etcd:4001/health &&
-    await-health-check http://mesos:5050/health &&
-    await-health-check -t=30 http://apiserver:8888/healthz &&
+    await-health-check http://mesosmaster1:5050/health &&
+    await-health-check -t=60 http://apiserver:8888/healthz &&
     km controller-manager
     --master=http://apiserver:8888
-    --cloud_config=/opt/mesos-cloud.conf
+    --cloud-config=/opt/mesos-cloud.conf
     --v=2
   links:
   - etcd
```

Comment:
In this case, it's using the etcd hostname for the first await-health-check. I could remove that check and just await apiserver, but i'd have to set the timeout to like 2 minutes.